service/storage_proxy.cc

```diff
@@ -3897,7 +3895,8 @@ template<typename Range>
 bool storage_proxy::cannot_hint(const Range& targets, db::write_type type) const {
     // if hints are disabled we "can always hint" since there's going to be no hint generated in this case
     return hints_enabled(type) &&
-            _hints_manager.started() &&
+            _hints_manager.started() && // If the manager hasn't started yet, no mutation will be performed to another node.
+                                        // No hint will need to be stored.
```

Comment:
I'm all for incremental improvements and cleanups, but this one is completely unrelated to the topic of this PR. Could you send it as a separate pull request?